beacon-chain/state/state-native/getters_payload_header.go

```diff
@@ -19,11 +21,20 @@ func (b *BeaconState) LatestExecutionPayloadHeader() (*enginev1.ExecutionPayload
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 
-	return b.latestExecutionPayloadHeaderVal(), nil
+	if b.version == version.Bellatrix {
```

Comment:
I think in general and considering future forks, the best approach for all these switches is the opposite order
```
if version >= Capella { do something  }
else if version >= Bellatrix { do something }
else { not supported }
```
The first one is hit more often.